assets/app/features/upgrade/upgrade.controller.spec.js

```diff
@@ -44,14 +54,51 @@ describe('Upgrade Controller', function () {
             should.exist(controller.steps);
         });
 
-        it('should contains a nextStep function ', function() {
-            should.exist(controller.steps.nextStep);
-            expect(controller.steps.nextStep).toEqual(jasmine.any(Function));
+        describe('nextStep function', function () {
+            it('should exist', function () {
+                should.exist(controller.steps.nextStep);
+                expect(controller.steps.nextStep).toEqual(jasmine.any(Function));
+            });
+
+            it('when called it should move to the next step', function () {
+                expect($state.current.id).toBe(0);
+                // calculate the next step in the list
+                var nextStep = (controller.steps.list[upgradeStepsFactory.activeStep.id + 1]).state;
+                controller.steps.nextStep();
+                expect($state.go).toHaveBeenCalledWith(nextStep);
+            });
+
+            it('when called in the last step it should not change steps', function () {
+                // set the last step in the list as current
+                $state.current = {id: 0, name: 'upgrade.upgrade-nodes', state: 'upgrade.upgrade-nodes'};
+                // refresh the list so it sets it as active
+                upgradeStepsFactory.refeshStepsList();
```

Comment:
I'd even consider going one step further and mock the response for `isLastStep()` function to return true/false according to the scenario being tested, and ignore completely how the last step is being determined.